lib/apiservers/engine/backends/container.go

```diff
@@ -1349,6 +1362,38 @@ func setPathFromImageConfig(config, imageConfig *containertypes.Config) {
 func validateCreateConfig(config *types.ContainerCreateConfig) error {
 	defer trace.End(trace.Begin("Container.validateCreateConfig"))
 
+	// process cpucount here
+	var cpuCount int64 = DefaultCPUs
+	if config.HostConfig.CpusetCpus != "" {
+		cpus := strings.Split(config.HostConfig.CpusetCpus, ",")
+		if c, err := strconv.Atoi(cpus[0]); err == nil {
+			cpuCount = int64(c)
+		} else {
+			log.Warnf("Error parsing CPU count, using default of %d: %s", DefaultCPUs, err)
```

Comment:
This should prevent container creation and return a remediable error to the user.
Generally a container configuration is pass/fail; while we supply defaults we should not be attempting to provide a sort-of-like-what-you-asked-for if there were specifics that couldn't be fulfilled.

The only places where that isn't true that I can think of are floor/ceiling constraints where we'd be requiring the user to parrot system internal limitations. There a warning should be provided to the user, but we can adjust and continue.